lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_lowering_context.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <torch/jit.h>
+
+#include "lazy_tensor_core/csrc/lowering_context.h"
+
+namespace torch_lazy_tensors {
+namespace compiler {
+
+using TSOpVector = std::vector<torch::jit::Value*>;
+
+namespace ts_backend {
+
+class GenericComputationTS : public lazy_tensors::GenericComputation {
+ public:
+  GenericComputationTS(std::shared_ptr<torch::jit::Graph> graph)
+      : graph_(std::move(graph)) {}
+
+  lazy_tensors::StatusOr<lazy_tensors::ProgramShape> GetProgramShape()
+      const override {
+    std::vector<std::string> parameter_names;
+    for (torch::jit::Value* input : graph_->inputs()) {
+      parameter_names.push_back(input->debugName());
+    }
+    // TODO(asuhan): should we use valid shapes?
```

Comment:
what does this mean?

Reply:
The program shape could be used by the compiler back-end for various purposes, such as assigning a certain physical layout - that's what happens with the TPU back-end, for example. I've checked and I'm pretty sure we're not using anywhere else, so for this particular back-end it shouldn't matter. I think I've answered my own TODO and I should add this observation instead - we don't need the actual result shape here.